### PR TITLE
Feature/lxl 4209 allow outgoing links to deleted

### PR DESCRIPTION
**Pull request by kwahlin**
See https://jira.kb.se/browse/LXL-4209. Third try!

### CODE REVIEW

File: whelk-core/src/main/groovy/whelk/filter/LinkFinder.groovy [hunk 1]
```diff
@@ -114,11 +114,19 @@ class LinkFinder {
...
         if (id != null && data.keySet().size() == 1) {
-            String primaryId = lookupPrimaryId(id)
+            // Path to same form as in lddb__dependencies.relation
+            String normalizedPath = (path.take(2) == Document.recordPath
+                    ? [JsonLd.RECORD_KEY] + path.drop(2)
+                    : (path.take(2) == Document.thingPath ? path.drop(2) : path)
+            )
+                    .findAll { it instanceof String }
+                    .join('.')
+            String primaryId = lookupPrimaryId(id, normalizedPath)
```
Comment (niklasl) [inline whelk-core/src/main/groovy/whelk/filter/LinkFinder.groovy:129]: Just so I can grasp the context; what was failing here before?

Comment (kwahlin) [reply whelk-core/src/main/groovy/whelk/filter/LinkFinder.groovy:129]: Nothing was failing before, however `lookupPrimaryId()` would throw `LinkValidationException` regardless of which relation links to a deleted record.

My previous attempts at https://jira.kb.se/browse/LXL-4209 only fixed the opposite case, i.e. enabling deletion of records with _incoming_ links via certain relations. This new fix is for allowing to delete/save a record with _outgoing_ links to deleted records via these relations.

Comment (kwahlin) [reply whelk-core/src/main/groovy/whelk/filter/LinkFinder.groovy:129]: Example:
`<WorkRecord> :derivedFrom <InstanceRecord1>, <InstanceRecord2>`
If <InstanceRecord2> is deleted, then you wouldn't be able to delete or resave <WorkRecord> due to the broken link.

Comment (niklasl) [reply whelk-core/src/main/groovy/whelk/filter/LinkFinder.groovy:129]: Ah; perfect; now I see it! Good work!

**Review by niklasl — APPROVED** [review]
LGTM! :ship:

**Pull request merged**